src/System.IO.Pipelines/src/System/IO/Pipelines/StreamPipeReader.cs

```diff
@@ -277,22 +292,14 @@ public override bool TryRead(out ReadResult result)
         {
             ThrowIfCompleted();
 
-            return TryReadInternal(InternalTokenSource, out result);
+            return TryReadInternal(out result);
         }
 
-        private bool TryReadInternal(CancellationTokenSource source, out ReadResult result)
+        private bool TryReadInternal(out ReadResult result)
         {
-            bool isCancellationRequested = source.IsCancellationRequested;
-            if (isCancellationRequested || _bufferedBytes > 0 && (!_examinedEverything || _isStreamCompleted))
+            if (_awaitable.IsCompleted || (_bufferedBytes > 0 && (_awaitable.IsCompleted || _isStreamCompleted)))
             {
-                if (isCancellationRequested)
-                {
-                    ClearCancellationToken();
-                }
-
-                ReadOnlySequence<byte> buffer = _readHead == null ? default : GetCurrentReadOnlySequence();
-
-                result = new ReadResult(buffer, isCancellationRequested, _isStreamCompleted);
+                result = GetReadResult();
```

Comment:
So how is StreamPipeReader.TryRead() supposed to work? If no one ever calls StreamPipeReader.ReadAsync() it just forever no-ops and returns false, right? That doesn't seem good.

And if you have to call StreamPipeReader.ReadAsync() anyway, why not just await the result? If you try not to await the result StreamPipeReader.ReadAsync() and then call StreamPipeReader.TryRead() until it returns true, not only is that inefficient it's likely to be racy since StreamPipeReader doesn't lock when reading or updating fields _readHead/Tail.